libnmstate/nm/connection.py

```diff
@@ -125,7 +125,8 @@ def nmdevice(self, dev):
 
     @property
     def con_id(self):
-        return self._con_id
+        con_id = self._con_profile.get_id() if self._con_profile else None
```

Comment:
This looks like a workaround and it appears to me that `self._con_id` should be properly initialized when `self._con_profile` is set. Or is there a use case where `self._con_id` needs to be different than `self._con_profile.get_id()`?

Reply:
There are multiple places where the profile is set (create, imports, explicitly setting it).
I see no problem in adding a simple logic when someone asks for this info: Either fetch the explicitly set value or read it from the profile itself.
Consider that a user of this object may ask about this ID even before the actual profile is created.